agol_connection.py

```diff
@@ -0,0 +1,119 @@
+from arcgis.gis import GIS
+from arcgis.features import FeatureLayerCollection, FeatureSet, Table, Feature
+import tempfile
+import shutil
+import json
+import csv
+import os
+
+
+class AGOLConnection(object):
+
+    def __init__(self):
+
+        creds = self._load_credentials()
+        if creds is None:
+            raise Exception("no arcgis credentials supplied")
+
+        self.creds = creds
+        self.layers = self._get_layers()
+        self.gis = self._get_gis()
+
+    def _load_credentials(self):
+
+        cred_path = "creds.csv"
+        if not os.path.isfile(cred_path):
+            return None
+
+        creds = {}
+        with open(cred_path, newline="") as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                if row['service'] == 'arcgis':
+                    creds['username'] = row['username']
+                    creds['password'] = row['password']
+                    creds['host'] = row['host']
+        return creds
+
+    def _get_layers(self):
+
+        with open("agol_layers.json", "r") as openf:
+            configs = json.loads(openf.read())
+
+        return configs
+
+    def _get_gis(self):
+        username = self.creds['username']
+        password = self.creds['password']
+        host = self.creds['host']
+
+        return GIS(host, username, password)
+
+    def get_arcgis_feature_collection_from_item_id(self, arcgis_item_id):
+
+        # You might ask - why do you not just use the FeatureLayerCollection's URL?
+        # Because you get a 403 if you try that. Instead, if you grab the parent container
+        # from the published layer, you can use the FLC manager's overwrite() method successfully.
+
+        feature_item = self.gis.content.get(arcgis_item_id)
+        if "type:Table" in str(feature_item):
+            fs = feature_item.tables[0].container
+        else:
+            fs = feature_item.layers[0].container
+        return fs
+
+    def upload_to_arcgis(self, dataset_name, source_data_dir, source_data_file):
```

Comment:
What do you think about renaming to `overwrite_arcgis_layer` or similar? That'd be more specific then `upload`.

Reply:
big +1 to that

Reply:
done